server/server.go

```diff
@@ -233,22 +238,14 @@ func (p *PackageIndexer) handleRequest(conn net.Conn) {
 		}
 
 		if Request.command == CmdRemove {
-			pkg, ok := p.Get(Request.pkg)
+			_, ok := p.Get(Request.pkg)
 			if !ok {
 				_, err := conn.Write([]byte(fmt.Sprintf("%s\n", ResponseOK)))
 				if err != nil {
 				}
 				continue
 			}
 
-			if len(pkg.dependents) > 0 && p.find(mapKeys(pkg.dependents)...) {
```

Comment:
@nickvanw the main change was to move this logic into the Remove function. The problem was that this block was calling "find" outside of any mutex protection.